modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java

```diff
@@ -585,6 +586,49 @@ public boolean isFormShared() {
 		return ParamUtil.getBoolean(_renderRequest, "shared");
 	}
 
+	public boolean isLimitToOneSubmissionPerUser() throws PortalException {
+		if (isDefaultUser() || !isLimitToOneSubmissionPerUserEnabled()) {
+			return Boolean.FALSE;
+		}
+
+		List<DDMFormInstanceRecordVersion> ddmFormInstanceRecordVersions =
+			_ddmFormInstanceRecordVersionLocalService.
+				getFormInstanceRecordVersions(getUserId(), getFormInstanceId());
+
+		Stream<DDMFormInstanceRecordVersion> stream =
+			ddmFormInstanceRecordVersions.stream();
+
+		List<DDMFormInstanceRecordVersion>
+			ddmFormInstanceRecordVersionsNotDrafts = stream.filter(
+				ddmFormInstanceRecordVersion ->
+					ddmFormInstanceRecordVersion.getStatus() !=
+						WorkflowConstants.STATUS_DRAFT
+			).collect(
+				Collectors.toList()
+			);
+
+		if (!ddmFormInstanceRecordVersionsNotDrafts.isEmpty()) {
+			return Boolean.TRUE;
+		}
+
+		return Boolean.FALSE;
+	}
+
+	public boolean isLimitToOneSubmissionPerUserEnabled()
```

Comment:
@marcelabc does this method need to be `public` or can we make it `private`?

Reply:
Yes, because I will need to use/access in a condition in other class in the next subtask.